merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MapperMethodMaker.java

```diff
@@ -175,15 +177,107 @@ public final MethodSpec makeGetValidationFailuresMethod() {
+  protected final MethodSpec.Builder makeArgumentAssignments(
+      final MethodSpec.Builder methodBuilder,
+      final BiFunction<CodeBlock.Builder, ParameterRecord, CodeBlock.Builder> makeArgumentAssignment)
+  {
+    var mb = methodBuilder;
+
+    // Condition must be checked for otherwise a try/catch without an exception thrown
+    // will not pass compilation
+    final var shouldExpectArguments = !exportType.parameters().isEmpty();
+
+    if (shouldExpectArguments) {
+      mb = mb
+          .addStatement(
+          "final var $L = new $T(\"$L\", \"$L\")",
+          "unconstructableArgsExBuilder",
+          UnconstructableArgumentsException.Builder.class,
+          metaName,
+          exportType.name());
+    }
+
+    mb = mb
+        .addStatement(
+            "final var $L = new $T(\"$L\", \"$L\")",
+            "extraneousArgsExBuilder",
+            ExtraneousArgumentsException.Builder.class,
+            metaName,
+            exportType.name())
+        .addCode("\n")
+        .beginControlFlow("for (final var $L : $L.entrySet())", "entry", "arguments");
+
+    if (shouldExpectArguments) {
+        mb = mb.beginControlFlow("try");
+    }
+
+    mb = mb
+        .beginControlFlow("switch ($L.getKey())", "entry")
+        .addCode(
+            exportType.parameters()
+                .stream()
+                .map(parameter -> {
+                    final var caseBuilder = CodeBlock.builder()
+                        .add("case $S:\n", parameter.name)
+                        .indent();
+                    return makeArgumentAssignment.apply(caseBuilder, parameter)
+                        .addStatement("break")
+                        .unindent();
+                })
+                .reduce(CodeBlock.builder(), (x, y) -> x.add(y.build()))
+                .build())
+        .addCode(
+            CodeBlock
+                .builder()
+                .add("default:\n")
+                .indent()
+                .addStatement(
+                    "$L.withExtraneousArgument($L.getKey())",
+                    "extraneousArgsExBuilder",
+                    "entry")
+                .unindent()
+                .build())
+        .endControlFlow();
+
+    if (shouldExpectArguments) {
+      mb = mb
+          .nextControlFlow("catch (final $T e)", UnconstructableArgumentException.class)
+          .addStatement(
+              "$L.withUnconstructableArgument(e.parameterName, e.failure)",
+              "unconstructableArgsExBuilder"
+          )
+          .endControlFlow();
+    }
+
+    mb = mb
+        .endControlFlow()
+        .addCode("\n");
+
+    if (shouldExpectArguments) {
+      mb = mb
+          .addStatement(
+              "$L.throwIfAny()",
+              "unconstructableArgsExBuilder");
+    }
+
+    mb = mb
+        .addStatement(
+            "$L.throwIfAny()",
+            "extraneousArgsExBuilder")
+        .addCode("\n");
```

Comment:
🔶 I'm wondering what we gain or lose from separating these three illegal argument exception types (Unconstructable, Extraneous, and Missing). 

One thing we lose is that we only ever throw one of these three, but the same activity could have all three simultaneously.

I'm wondering if collecting all of this information into one exception, and throwing it at the end, would allow us to both provide the maximum amount of feedback to users, and avoid cluttering the MissionModel interface with several related exception types.

Reply:
Very good point. I'm cool with going this direction! I'll rework things today to just throw one exception.

Reply:
This change has now been made :)